### ISSUE TITLE
Fails loading unqualified URLS with a path component

### ISSUE THREAD

**Issue by nicwaller** [blmayer/astro#19]
When I try opening a URL that is both unqualified (no `gemini://` prefix) and pathed (has `/astro.gmi` as a suffix) I find that astro fails to render the page. When I exit astro, I see this error in the terminal:

```
$ ./astro derelict.garden/astro.gmi
rm: cannot remove '/home/nic/.cache/astro/links': No such file or directory
```

It also affects URLs opened from within astro. Press `o` then `derelict.garden/astro.gmi`.

## Debug Mode

Running astro in debug mode yields some insight.

Passing case:

```
Starting with gemini://derelict.garden/astro.gmi
 - margin:     8
Parsing: gemini://derelict.garden/astro.gmi
Parsed URL: proto: gemini host: derelict.garden path: /astro.gmi
```

Failing case:

```
Starting with derelict.garden/astro.gmi
 - margin:     8
Parsing: derelict.garden/astro.gmi
Parsed URL: proto:  host:  path: derelict.garden/astro.gmi
Response: proto: gemini host: derelict.garden/astro.gmi port: 1965 path:
```

## Affected Versions

Reproducible for all versions versions from v0.8.0 onward. Most recently observed with v0.14.1 0642554dcced914c0dc0b319514ba3602e9b3015.

I went back to 0.7.2 (ddb79dd9d9d3eaabbc9496cbc6a75bdcc8726e97) and observed that the page rendered properly when invoked as shown above. The bug with this kind of URL seems to be present in all versions after that.

However, using the same version 0.7.2, I got an error trying to open `derelict.garden/astro.gmi` from within astro.

## Passing cases

For completeness, all of the following test cases work correctly and without error. ✅ 

```
./astro gemini://derelict.garden/astro.gmi
./astro gemini://derelict.garden
./astro derelict.garden
```

**Comment by blmayer** [comment]
Please take a look at #21. It passed your tests.

**Comment by nicwaller** [comment]
> Please take a look at #21. It passed your tests.

Thanks! I tested out a772b4a78fc3d94ef2640828935f1d193b33cba4 and confirmed that astro now loads and renders all the URLs I can throw at it. :) 

I still see an error message upon exit, but it doesn't seem to impact the experience of using the app, so it can probably be safely ignored for now.

```
nic@nicwaller:~/tmp/astro$ ./astro derelict.garden/astro.gmi
rm: cannot remove '2': No such file or directory
^C
nic@nicwaller:~/tmp/astro$
```

**Comment by blmayer** [comment]
I got that too, but `g+1` is *astro.png*, which doesn't exist on server, so it returns 51, and the previous page is loaded. So I'll fix it by uploading the image.

**Comment by blmayer** [comment]
You're right, I was trying the wrong link.

**Comment by blmayer** [comment]
There's a new commit on the fix, to fix the fix :-)

**Comment by blmayer** [comment]
Thanks @nicwaller for bringing attention to this bug. As the last commit passed your tests I plan to merge them and close this issue in the following day if you're OK with the solution.

**Comment by nicwaller** [comment]
> Thanks @nicwaller for bringing attention to this bug. As the last commit passed your tests I plan to merge them and close this issue in the following day if you're OK with the solution.

I took a look and posted [my feedback](https://github.com/blmayer/astro/pull/21#issuecomment-1168923917) on the PR thread. It seems not quite ready yet. 😐 

URL handling is really challenging, more than I expected! I think adding in some unit tests for URL handling might be warranted. Want me to send a PR with some unit tests?

**Comment by blmayer** [comment]
Thanks for the feedback on the PR.

> URL handling is really challenging, more than I expected! I think adding in some unit tests for URL handling might be warranted. Want me to send a PR with some unit tests?
Yes, that will be great.

**Comment by blmayer** [comment]
Finally fixed in #21. Thank you for the help. Feel free to open other issues if you find anything.

**Comment by nicwaller** [comment]
> Finally fixed in #21. Thank you for the help. Feel free to open other issues if you find anything.

🥂 Cheers! Happy to help.
